src/CliCommand/Mid3v2Command.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace IndieHD\AudioManipulator\CliCommand;
+
+use IndieHD\AudioManipulator\CliCommand\Mid3v2CommandInterface;
+
+class Mid3v2Command extends CliCommand implements Mid3v2CommandInterface
+{
+    protected $name = 'mid3v2_command';
+
+    protected $binary = 'mid3v2';
+
+    protected $parts = [
+        'options' => [],    // Global options
+        'infile' => [],     // Input file
+    ];
+
+    public function __construct()
+    {
+        if (!empty(getenv('MID3V2_BINARY'))) {
+            $this->binary = getenv('MID3V2_BINARY');
```

Comment:
This doesn't have to be done in this PR, However something to consider. Better practice would to create configuration file(s) and have those only get values from the `env` ... This way if a `env` variable has to change at some point for whatever reason, you only update the configuration files and not have to modify several implementations to update the env variable in `getenv()` ... Hopefully this makes sense.

Reply:
It does make sense, but my intention here was to allow for exactly what you describe.

It seems that any environment variable values should be set in the "parent" project that includes this library, and not in this library itself. In other words, the parent project could use something like https://github.com/vlucas/phpdotenv to set `MID3V2_BINARY=/home/foo/custom-mid3v2-binary` and the existing could would "just work", because `phpdotenv` would set the environment to reflect the values in the `.env` file.

Am I missing something here?

Reply:
So what happens when you decide that the variable `MID3V2_BINARY` now should be named something like `MID3V5_BINARY` ? ... you will have to `modify`  this file (and any other files) to reflect that change, where as you could simple update a config file and it will filter down to all inheriting files.

Also taking note to the filename itself (I am assuming that the name `Mid3v2` is formmated in the way of `Mid3v<version>` like it supports `v2.*`  ... note sure if this is something that could change in the future or not.

However, regarding the variable name above, maybe that's not why you change the variable name, but rather later determine that it makes better sense to name it something different ?

Reply:
I agree that it would be less than ideal to have to update that `MID3V2_BINARY` name in more than one place, should it ever need to be changed.

That current name is based only on the actual library/binary name, which is in turn named after the ID3v2` specification. The v2 refers to the specification at http://id3.org/id3v2.3.0 , and not the version of Mutagen (which happens to be the library that contains the binary we're using, `mid3v2`).

I think I see your point, now. You're simply suggesting that we make that a config file that remains local to this library, and not something that we intend to "publish" into the "parent" project, right? If that's the case, then yes, I concur, and we can do that in a subsequent PR.

Reply:
Mostly basing my opinion on the common practice that Laravel has implemented with environment variables.

https://github.com/alexeymezenin/laravel-best-practices#do-not-get-data-from-the-env-file-directly

Also is suggested by Laravel Shift, here is a related post https://laracasts.com/discuss/channels/general-discussion/laravel-shift-upgrade-to-52-use-config-instead-of-env

Reply:
We'll have to give this further consideration and not implement anything on a whim.

I say that for two reasons, primarily:

1. I don't want this package to be coupled to Laravel, which means that any generic solution would likely need an additional "bridge" package to make it work seamlessly with Laravel (or any other framework that we intend to support).
2. The `.env` file and `config()` helper seem like they belong in the parent framework/app, and not in this (or any other) "library" that is included as a dependency thereto. That's not to say that we shouldn't find a way to avoid "hard-coding" the environment variable names (e.g., `MID3V2_BINARY`) within the implementing classes; I'm just not sure that an env/config package of some kind solves that problem elegantly, given our specific use-case here.

Let's open a separate Issue and/or PR for that effort, regardless.